### DISCUSSION TITLE
Connection to energy dashboard

### DISCUSSION THREAD

**Discussion by danieleproietti** [tolwi/hassio-ecoflow-cloud#156] [Q&A]
Hi, I configured my powerstream correctly but I can't connect it to the energy dashboard, can you help me? Thank you

**Comment by Fyr42** [comment]
To edit the Energy Dashboard :
Settings -> Dashboard -> Energy

On the Solar Panel Section you have to build Helpers : Settings -> Devices and Services -> Helpers 

For each panel you create a new helper : Clic "+ Add Helpers" button  at bottom right and choose  "Integration - Riemann Sum Integral Sensor" 
Name it, In sensor choose powerstream_solar_1_watts or sensor.powerstream_solar_2_watts, 
Left Riemann
Precision you want 2 or 3
Metric is k (ilo)
Time unit : let it at hour

Now you have for each panel a new helper in KWh. And now put your new helpers  in your Energy dashboard in Solar section.

The same for the battery you could have. In my case I have a Delta 2Max so the correct sensors for the RIemann Sum is delta_2_max_total_in_power and delta_2_max_total_out_power, and you but the helpers you build in the battery section. You can'y use the sensor powerstream_battery_input_watts because it's positive or negative , depending if you are charging or draining so the integral is around zero at the end of the day. But it's good for instant value graph.

You have to wait a few minutes that the sensors start collect of few value. You can't manipulate a zero or bad unit sensor/helper.

**Reply by danieleproietti** [reply]
Hi,
you were really kind. So I have no problem with solar but I don't see data from the battery. I have an ecoflow delta 2 extra battery.
I configured both the powerstream and the battery.
I attach screenshots of the configuration.
Thank you

<img width="1021" alt="Screenshot 2023-09-24 alle 11 53 35" src="https://github.com/tolwi/hassio-ecoflow-cloud/assets/16803681/2e58ba71-62be-486b-af5f-0cb4070f701b">
<img width="1680" alt="Screenshot 2023-09-24 alle 11 55 06" src="https://github.com/tolwi/hassio-ecoflow-cloud/assets/16803681/a007b1c3-71a5-4f8b-8cee-afd16fc19fa7">
<img width="1680" alt="Screenshot 2023-09-24 alle 11 56 49" src="https://github.com/tolwi/hassio-ecoflow-cloud/assets/16803681/c16eca18-5301-429f-a262-c81eb60cd136">

**Comment by Fyr42** [comment]
It's ok for the value from you battery. Extra battery doesn't have an inverter inside so no value about it. Some part in the list are for the "main" battery model, like the first "battery temperature". Extra battery is just for storage not for AC <-> DC conversion, it's done by the PowerStream. You dont have AC output sockets too, so no value.

**Reply by danieleproietti** [reply]
However, I don't have any main battery, only the additional one. The EcoFlow system can also be used in this way, saving some money.
Is there a way I can extract the battery charging and discharging values?

**Reply by Fyr42** [reply]
Yes. Via the PowerStream. But not easily  in the Energy dashboard with the sensor powerstream_battery_input_watts It's negative when charging and positive when discharging in your home. 

So you have to make a condition to do a Riemann sum of only positive values in case you want it in the Energy dashboard. You can use it in an other card, called "Power Flow Card Plus" via HACS in Frontend, and display instant value of charge/discharge.
<img width="482" alt="Screenshot 2023-09-24 at 15 47 39" src="https://github.com/tolwi/hassio-ecoflow-cloud/assets/119723794/469acc8f-853f-43e3-9bda-400fbad08fa4"> The picture uses this sensor.

**Reply by danieleproietti** [reply]
OK all clear. Last question: does the Riemann sum update every 24 hours? That is, add the data for the day and then reset? Thank you

**Reply by Fyr42** [reply]
Both cards have a daily cycle. 

For the Riemann sum the last value is the new zero.
For the +/- sensor, put "PowerStream Battery Input Watts"  straight in the flow card under battery, the daily cycle will do it for you.

**Comment by danieleproietti** [comment]
I'm close... I used this code:

`
sensor:
  - platform: integration
    source: sensor.ecoflow_powerstream_battery_input_watts
    name: "Uscita batteria"
    unit_prefix: k
    round: 2
    method: right

  - platform: integration
    source: sensor.ecoflow_powerstream_battery_input_watts
    name: "Entrata batteria"
    unit_prefix: k
    round: 2
    method: left

  - platform: template
    sensors:
      entrata_batteria_absolute:
        value_template: "{{ state_attr('sensor.entrata_batteria', 'state') | float | abs }}"
        unit_of_measurement: "kWh"
        friendly_name: "Entrata batteria (assoluto)"
`

adding the "entrata_batteria_absolute" template to ensure that the battery input values are always positive but the energy dashboard cannot find the template 😭

**Reply by Fyr42** [reply]
The energy dashboard control that the unit of the sensor is usable. You really need kilo-watt-hour. The "unit of measurement" is just a text label, not a physical unit needed by the dashboard.

**Reply by danieleproietti** [reply]
I didn't explain myself well.
I used a Riemann integral sum with "right" method to take the values when the battery discharging in my home

Then I used a Riemann integral sum with "left" method to take the values when charging the battery

Then I took these two sensors and used them in the energy dashboard but at the value "battery output" it tells me: "The entity has a negative state" (screenshot)

So I created a template to always transform that sensor into a positive one but this sensor template does not appear in the energy dashboard, I cannot select it 😤

<img width="415" alt="Screenshot 2023-09-25 alle 16 29 35" src="https://github.com/tolwi/hassio-ecoflow-cloud/assets/16803681/d55d8cf1-daa2-4167-b23a-72523b9246f4">

**Reply by Fyr42** [reply]
looklike there is a negative value. Go in "Developper Tools" -> "Statistics" on the right clic the "stairs" at your sensors. Browse the values and correct to positive or put zero.
